src/content/en/showcase/2017/within.md

```diff
@@ -0,0 +1,386 @@
+the Vive controllers. The hitbox's size auto updates when you change the text
+options.
+
+Bmtext.group is added to the three.js scene. If you need to access the children
+/ Object3D's, the scene graph for the text looks like:
+
+<img src="/web/showcase/2017/images/within/diagram2.png">
+
+## 6. Unminify json and modify xoffsets
+
+<img src="/web/showcase/2017/images/within/8.gif">
+
+
+If your kernings look off, you may need to edit the xoffsets in the json. Paste
+the json into [Jsbeautifier.org](http://jsbeautifier.org/)** **to get an
+unminified version of the file.
+
+The xoffset is essentially global kerning for one character. Kerning is for two
+specific characters that appear next to each other. The default values in the
+kernings array don't actually make a difference, and it would be too tedious to
+edit, so you can empty that array to decrease the file size of the json. Then
+edit the xoffsets for kerning. But first you'll have to figure out which
+characters go with which char id in the json.
+
+In [three-bmfont-text-bundle.js](https://vr.with.in/archive/text-sdf-bitmap/three-bmfont-text-bundle.js), insert a console.log after line 240:
+
+    var id = text.charCodeAt(i)
+    // console.log(id);
+
+Then type into dat.gui text field on
+[https://vr.with.in/archive/text-sdf-bitmap/](https://vr.with.in/archive/text-sdf-bitmap/)
+and check the console to find the corresponding id of a character.
+
+For example, in our bitmap font, "j" is consistently too far to the right. Its
+char id is 106. So find "id": 106 in the json and change its xoffset from -1 to
+-10.
+
+## 7. Layout
+
+If you have multiple blocks of text and want it to flow from top to bottom like
+HTML, everything has to be manually positioned, similar to absolute positioning
+every dom element yourself with CSS. Can you imagine doing this in CSS?
+
+    * { position: absolute; }
+
+That's what text layout in 3D is like. In the detail view: title, author,
+description, and duration are each a new TextBitmap object with their own
+styles, color, scale, etc.:
+
+
+<img src="/web/showcase/2017/images/within/9.jpg">
+
+    author.group.position.y = title.group.position.y - title.height - padding;
+    description.group.position.y = author.group.position.y - author.height - padding;
+    duration.group.position.y = description.group.position.y - description.height - padding;
+
+This assumes that the local origin of each TextBitmap group is vertically
+aligned with the top of the TextBitmap mesh (see centering in
+[text-bitmap.js](https://vr.with.in/archive/text-sdf-bitmap/text-bitmap.js)
+update). If you change the text for any of those objects later, and the height
+of that object changes, you will also need to recalculate those positions. Here
+only the y-position of the text is modified, but one opportunity of working in
+3D is that we can push and pull the text in the z-direction, as well rotate
+around the x, y and z axis.
```

Comment:
The plural of axis is axes. Would have changed it myself, but want to make sure you know it's not a mistake.